modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageExpirationWorker.java

```diff
@@ -62,8 +81,72 @@ public void run() {
+                                long currentDeletionRangeLowerBoundId = messageIDSet.
+                                        toArray(new Long[messageIDSet.size()])[safetySlotCount - 1] + 1;
+
+                                //get the expired messages for that queue in the range of message ID starting form the lower bound ID
+                                List<AndesMessageMetadata> expiredMessages = MessagingEngine.getInstance()
+                                        .getExpiredMessages(currentDeletionRangeLowerBoundId, queueName);
+
+                                if (null != expiredMessages && !expiredMessages.isEmpty()) {
+
+                                    //in order to stop the slot delivery worker to this queue in the deletion range
+                                    SlotManagerClusterMode.setCurrentDeletionQueue(queueName);
+                                    SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+                                    SlotManagerStandalone.setCurrentDeletionQueue(queueName);
+                                    SlotManagerStandalone.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+
+                                    //delete message metadata, content from the db
+                                    MessagingEngine.getInstance().deleteMessages(expiredMessages);
+
+                                    //since the deletion task is finished that queue no need to hold the slot delivery worker for that queue
+                                    SlotManagerClusterMode.setCurrentDeletionQueue("");
+                                    SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(0L);
+                                    SlotManagerStandalone.setCurrentDeletionQueue("");
+                                    SlotManagerStandalone.setCurrentDeletionRangelowerboundID(0L);
+
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Expired message count for queue : " + queueName + "is" + expiredMessages.size());
+                                    }
+
+                                }
+
+                            }
+
+                        }
+
+                    }
+                    //sleep the message expiration worker for specified amount of time
+                    sleepForWaitInterval(workerWaitInterval);
+
+                    /*
                     //Get Expired message IDs from the database with the massageBatchSize as the limit
-                    // we cannot delegate a cascaded delete to cassandra since it doesn't maintain associations between columnfamilies.
+                    // we cannot delegate a cascaded delete to cassandra since it doesn't maintain associations between column families.
                     List<AndesMessageMetadata> expiredMessages = MessagingEngine.getInstance().getExpiredMessages(messageBatchSize);
```

Comment:
I think this will generate a compile time error. `getExpiredMessages` method accepts two parameters but only one is provided here.